[Bug] uart.c has wrong arguments

<!-- Provide a general summary of the bug in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Describe the Bug

The `uart_write()` function in `uart.c` had a wrong argument (`c` instead of `data`). 
During compilation the error was
```
Compiling: platforms/chibios/drivers/uart.c                                                        In file included from ./lib/chibios/os/hal/include/hal.h:136,
                 from platforms/chibios/drivers/uart.h:21,
                 from platforms/chibios/drivers/uart.c:17:
platforms/chibios/drivers/uart.c: In function 'uart_write':
platforms/chibios/drivers/uart.c:46:55: error: 'c' undeclared (first use in this function); did you mean 'ch'?
   46 | void uart_write(uint8_t data) { sdPut(&SERIAL_DRIVER, c); }
        |      
```
Manually changing the argument solved the issue. The bug originated from commit `04b51e381e2ff3f4c7aba662e1c817ce5daa931d`.

## System Information

 - QMK Firmware version: 0.15.11
<!-- Run `git describe --abbrev=0 --tags` to find this out. -->

--- a/platforms/chibios/drivers/uart.c
+++ b/platforms/chibios/drivers/uart.c
@@ -44,7 +44,7 @@ void uart_init(uint32_t baud) {
 }
 
 void uart_write(uint8_t data) {
-    sdPut(&SERIAL_DRIVER, c);
+    sdPut(&SERIAL_DRIVER, data);
 }
 
 uint8_t uart_read(void) {

